JASP-Engine/JASP/R/correlation.R

```diff
@@ -15,806 +15,721 @@
 
-				}
+		if (length(options$variables) < 2) {
 
-				plot <- list()
+			return(list(results=results, status="complete", keep=keep))
 
-				plot[["title"]] <- "Correlation Plot"
-				plot[["width"]]  <- width
-				plot[["height"]] <- height
+		} else {
 
-				correlation.plot <- plot
-			}
+			return(list(results=results, status="inited", state=state, keep=keep))
 		}
-	}
-
-	if (perform == "run" && length(options$variables) > 0 && options$plotCorrelationMatrix) {
-
 
-		if (!is.null(state) && !is.null(diff) && ((is.logical(diff) && diff == FALSE) || (is.list(diff) && (diff$confidenceIntervals == FALSE && diff$confidenceIntervalsInterval == FALSE
-			&& diff$hypothesis == FALSE && diff$kendallsTauB == FALSE && diff$missingValues == FALSE && diff$pearson == FALSE && diff$plotCorrelationMatrix == FALSE
-			&& diff$plotDensities == FALSE && diff$plotStatistics == FALSE && diff$spearman == FALSE && diff$variables == FALSE)))) {
-
-			# if only "Report significance", "Flag significant correlations", "Means and Standard Deviations" or "Cross-product deviations and covariances" have changed, the previous plot can be used
+	} else { # run
+    
+    state <- list(options=options, correlationPlot=correlation.plot, tableData=tableData)
+    attr(state, "key") <- stateKey
+    
+		return(list(results=results, status="complete", state=state, keep=keep))
+	}
+}
 
-			correlation.plot <- state$correlationPlot
+.calculateCorrelations <- function(dataset, perform, options, stateTableData, variables=c(), pearson=TRUE, kendallsTauB=FALSE,
+	spearman=FALSE, hypothesis="correlated", CI=0.95) {
+  
+  ready <- FALSE
+  variables <- unlist(variables)
+  if (length(variables) == 0) {
+    variables <- c(variables, "...", "... ") # we need this trailing space so 1 != 2
+  } else if (length(variables) == 1) {
+    variables <- c(variables, "...")
+  } else {
+    ready <- TRUE
+  }
+  
+	tests <- c()
+	if (pearson)
+		tests <- c(tests, "pearson")
+	if (spearman)
+		tests <- c(tests, "spearman")
+	if (kendallsTauB)
+		tests <- c(tests, "kendall")
 
-		} else {
+  results <- list(tests=tests, variables=variables, hypothesis=hypothesis, CI=CI)
+  
+  if (length(tests) == 0) {
+    return(results)
+  }
+  
+  pairs <- combn(variables, 2, simplify=FALSE)
+	for (i in seq_along(pairs)) {
 
-			variables <- unlist(options$variables)
+		var1 <- pairs[[i]][1]
+    var2 <- pairs[[i]][2]
+    pairName <- paste(sort(c(var1, var2)), collapse="-")
 
-			# check variables
-			d <- vector("character", length(.v(variables)))
-			sdCheck <- vector("numeric", length(.v(variables)))
-			infCheck <- vector("logical", length(.v(variables)))
+		for (test in tests) {
+    
+      errorMessage <- NULL
+      estimate <- p.value <- MPR <- upperCI <- lowerCI <- "."
+      
+      if (! is.null(stateTableData) && ! is.null(stateTableData[["result"]][[pairName]][[test]])) {
 
-			for (i in seq_along(.v(variables))) {
+        state <- stateTableData[["result"]][[pairName]][[test]]
```

Comment:
This state refers to a retrieved object rather than state in general. A name such as corObj would be better here